src/image.rs

```diff
@@ -366,27 +362,21 @@ pub trait ImageDecoder<'a>: Sized {
     fn dimensions(&self) -> (u64, u64);
 
     /// Returns the color type of the image data produced by this decoder
-    fn colortype(&self) -> ColorType;
+    fn color_type(&self) -> ColorType;
 
     /// Retuns the color type of the image file before decoding
-    fn original_colortype(&self) -> ExtendedColorType {
-        self.colortype().into()
+    fn original_color_type(&self) -> ExtendedColorType {
+        self.color_type().into()
     }
 
     /// Returns a reader that can be used to obtain the bytes of the image. For the best
     /// performance, always try to read at least `scanline_bytes` from the reader at a time. Reading
     /// fewer bytes will cause the reader to perform internal buffering.
     fn into_reader(self) -> ImageResult<Self::Reader>;
 
-    /// Returns the number of bytes in a single row of the image. All decoders will pad image rows
-    /// to a byte boundary.
-    fn row_bytes(&self) -> u64 {
-        (self.dimensions().0 * color::bits_per_pixel(self.colortype()) as u64 + 7) / 8
-    }
-
-    /// Returns the total number of bytes in the image.
+    /// Returns the total number of bytes in the decoded image.
     fn total_bytes(&self) -> u64 {
-        self.dimensions().1 * self.row_bytes()
+        self.dimensions().0 * self.dimensions().1 * self.color_type().bytes_per_pixel() as u64
```

Comment:
Should have some justification of overflow behaviour. I believe it is that it should not be possible for a valid image? But I'm not sure, this has been a weak spot for a long time but it may be okay.

Reply:
Is there a library-wide restriction on width and height?

Reply:
This can theoretically overflow, but only if the decoder's reported dimensions + color type indicated a decoded size of greater than 16 EB. That doesn't remotely come close to fitting in memory any time in the foreseeable future, so I'm really not too concerned about it. If you'd prefer I can change it to using checked multiplication instead though.